OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromDiscLogic.cs

```diff
@@ -486,11 +486,17 @@ string FindSourcePath(ModContent.ModSource source, IEnumerable<string> volumes)
 
 				foreach (var prefix in source.RegistryPrefixes)
 				{
-					var path = Microsoft.Win32.Registry.GetValue(prefix + source.RegistryKey, source.RegistryValue, null) as string;
-					if (path == null)
+					var value = Microsoft.Win32.Registry.GetValue(prefix + source.RegistryKey, source.RegistryValue, null) as string;
+					if (value == null)
 						continue;
 
-					return IsValidSourcePath(path, source) ? path : null;
+					if (source.Type == ModContent.SourceType.RegistryDirectoryFromFile)
```

Comment:
We can avoid duplication here by keeping the variable name as `path` and then just doing
```csharp
if (source.Type == ModContent.SourceType.RegistryDirectoryFromFile)
	path = Path.GetDirectoryName(path);
```
before the return.